### PR TITLE
add InvalidateBlock RPC command

### PR DESCRIPTION
**Pull request by rubensayshi**

### CODE REVIEW

File: chain.go [hunk 1]
```diff
@@ -749,3 +749,38 @@ func (c *Client) RescanBlocksAsync(blockHashes []chainhash.Hash) FutureRescanBlo
 func (c *Client) RescanBlocks(blockHashes []chainhash.Hash) ([]btcjson.RescannedBlock, error) {
 	return c.RescanBlocksAsync(blockHashes).Receive()
 }
+
+// FutureGetBlockResult is a future promise to deliver the result of a
```
Comment (davecgh) [inline chain.go]: s/FutureGetBlockResult /FutureInvalidateBlockResult/

File: chain.go [hunk 2]
```diff
@@ -749,3 +749,38 @@ func (c *Client) RescanBlocksAsync(blockHashes []chainhash.Hash) FutureRescanBlo
...
+// GetBlockAsync RPC invocation (or an applicable error).
```
Comment (davecgh) [inline chain.go]: s/GetBlockAsync /InvalidateBlockAsync/

File: chain.go [hunk 3]
```diff
@@ -749,3 +749,38 @@ func (c *Client) RescanBlocksAsync(blockHashes []chainhash.Hash) FutureRescanBlo
...
+	hash := ""
+	if blockHash != nil {
+		hash = blockHash.String()
+	}
+
+	cmd := btcjson.NewInvalidateBlockCmd(hash)
+	return c.sendCmd(cmd)
+}
+
+// InvalidateBlock returns a raw block from the server given its hash.
```
Comment (davecgh) [inline chain.go]: It looks like this comment was updated since it's inaccurate.

File: chain.go [hunk 4]
```diff
@@ -749,3 +749,38 @@ func (c *Client) RescanBlocksAsync(blockHashes []chainhash.Hash) FutureRescanBlo
...
+//
+// See InvalidateBlockVerbose to retrieve a data structure with information about the
```
Comment (davecgh) [inline chain.go]: This doesn't apply since there is no `InvalidateBlockVerbose`.

**Comment by rubensayshi** [comment]
Sorry, very sloppy of me to not look at the comments ...

**Pull request merged**